internal/expr/parser.go

```diff
@@ -265,28 +272,174 @@ func (p *Parser) parseIdentifier() (string, bool) {
 	return "", false
 }
 
+// parseColumn parses a column made up of name bytes, optionally dot-prefixed by
+// its table name. parseColumn returns an error so that it can be used with
+// parseList.
+func (p *Parser) parseColumn() (fullName, bool, error) {
+	cp := p.save()
+
+	if id, ok := p.parseIdentifierAsterisk(); ok {
+		if id != "*" && p.skipByte('.') {
+			if idCol, ok := p.parseIdentifierAsterisk(); ok {
+				return fullName{prefix: id, name: idCol}, true, nil
+			}
+		} else {
+			// A column name specified without a table prefix should be in name.
+			return fullName{name: id}, true, nil
+		}
+	}
+
+	cp.restore()
+	return fullName{}, false, nil
+}
+
+func (p *Parser) parseTarget() (fullName, bool, error) {
+	cp := p.save()
+
+	if p.skipByte('&') {
+		return p.parseGoFullName()
+	}
+
+	cp.restore()
+	return fullName{}, false, nil
+}
+
 // parseGoFullName parses a Go type name qualified by a tag name (or asterisk)
-// of the form "TypeName.col_name". On success it returns the parsed fullName,
+// of the form "&TypeName.col_name". On success it returns the parsed FullName,
```

Comment:
Curious.. this one was already lowercased, and should probably stay that way.

Reply:
Done.